test/func_tests/gs_tests.cpp

```diff
@@ -134,6 +137,46 @@ void print_goalstateReply(GoalStateOperationReply gsOperationReply)
   g_total_ACA_Message_time += gsOperationReply.message_total_operation_time();
 }
 
+// Synchronous server implementation to test the grpc client's connectivity.
+class GoalStateProvisionerServer final : public GoalStateProvisioner::Service {
+  grpc::Status RequestGoalStates(ServerContext *ctx, const HostRequest *request,
+                                 HostRequestReply *response) override
+  {
+    ctx->client_metadata();
+    request->CheckInitialized();
+    ACA_LOG_INFO("%s", "Test Server code called!");
+    response->mutable_operation_statuses()->Add();
+    response->mutable_operation_statuses()->at(0).set_operation_status(OperationStatus::SUCCESS);
```

Comment:
Beside operation status, we also want to verify resource id.

Reply:
Fixed